bundles/automation/org.eclipse.smarthome.automation.module.media/src/main/java/org/eclipse/smarthome/automation/module/media/internal/MediaActionTypeProvider.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2016 Kai Kreuzer and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.automation.module.media.internal;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.automation.Visibility;
+import org.eclipse.smarthome.automation.module.media.handler.PlayActionHandler;
+import org.eclipse.smarthome.automation.type.ActionType;
+import org.eclipse.smarthome.automation.type.Input;
+import org.eclipse.smarthome.automation.type.ModuleType;
+import org.eclipse.smarthome.automation.type.ModuleTypeProvider;
+import org.eclipse.smarthome.automation.type.Output;
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter.Type;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameterBuilder;
+import org.eclipse.smarthome.config.core.ParameterOption;
+import org.eclipse.smarthome.core.audio.AudioManager;
+
+/**
+ * This class dynamically provides the Play action type.
+ * This is necessary since there is no other way to provide dynamic config param options for module types.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
+ */
+public class MediaActionTypeProvider implements ModuleTypeProvider {
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public ModuleType getModuleType(String UID, Locale locale) {
+        if ("PlayAction".equals(UID)) {
+            return getPlayActionType(locale);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public Collection<ModuleType> getModuleTypes(Locale locale) {
+        return Collections.singleton(getPlayActionType(locale));
+    }
+
+    private ModuleType getPlayActionType(Locale locale) {
+        return new ActionType("PlayAction", getConfigDesc(), "Play Sound", "Plays a sound file", null,
```

Comment:
The whole point about this class is to construct the type (specifically its config parameter options) dynamically on every new request.